File: Algorithms/Migratory Birds/migratory_birds.py
import math
import os
import random
import re
import sys

# Complete the migratoryBirds function below.
def migratoryBirds(arr):
	arr.sort()
	repeat = [0, 0, 0, 0, 0]

	for spot in arr:
		repeat[spot-1] += 1

	max = repeat[0]
	most = 1

	for index in range(1, len(repeat)):
		if repeat[index] > max:
			max = repeat[index]
			most = index + 1

	return most


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    arr_count = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    result = migratoryBirds(arr)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Algorithms/Implementation/Migratory Birds/migratory_birds.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the migratoryBirds function below.
function migratoryBirds(arr) {
	var birds = Array(5).fill(0);
	var max = 0;
	var most = 0;

	for (var spot in birds) {
		birds[spot] = arr.filter(a => a == parseInt(spot)+1).length;
	}

	for (var spot in birds) {
		if (birds[spot] > max) {
			max = birds[spot];
			most = parseInt(spot) + 1;
		}
	}
	return most;
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const arrCount = parseInt(readLine().trim(), 10);

    const arr = readLine().replace(/\s+$/g, '').split(' ').map(arrTemp => parseInt(arrTemp, 10));

    const result = migratoryBirds(arr);

    ws.write(result + '\n');

    ws.end();
}
